mobius-android/src/test/java/com/spotify/mobius/android/MobiusLoopViewModelTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+import androidx.lifecycle.Lifecycle;
+import com.spotify.mobius.First;
+import com.spotify.mobius.Mobius;
+import com.spotify.mobius.Next;
+import com.spotify.mobius.Update;
+import com.spotify.mobius.android.MobiusLoopViewModelTestUtilClasses.TestEffect;
+import com.spotify.mobius.android.MobiusLoopViewModelTestUtilClasses.TestEvent;
+import com.spotify.mobius.android.MobiusLoopViewModelTestUtilClasses.TestModel;
+import com.spotify.mobius.android.MobiusLoopViewModelTestUtilClasses.TestViewEffect;
+import com.spotify.mobius.android.MobiusLoopViewModelTestUtilClasses.TestViewEffectHandler;
+import com.spotify.mobius.functions.Consumer;
+import com.spotify.mobius.runners.ImmediateWorkRunner;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MobiusLoopViewModelTest {
+
+  private final List<TestEvent> recordedEvents = new LinkedList<>();
+  private final List<TestModel> recordedModels = new LinkedList<>();
+  private final Update<TestModel, TestEvent, TestEffect> updateFunction =
+      (model, event) -> {
+        recordedModels.add(model);
+        recordedEvents.add(event);
+        return Next.noChange();
+      };
+  @Rule public InstantTaskExecutorRule rule = new InstantTaskExecutorRule();
+  private MobiusLoopViewModel<TestModel, TestEvent, TestEffect, TestViewEffect> underTest;
+  private TestViewEffectHandler<TestEvent, TestEffect, TestViewEffect> testViewEffectHandler;
+
+  private FakeLifecycleOwner fakeLifecycle;
+  private RecordingObserver<TestModel> recordingModelObserver = new RecordingObserver<>();
+  private RecordingObserver<TestViewEffect> recordingForegroundViewEffectObserver =
+      new RecordingObserver<>();
+  private RecordingObserver<Iterable<TestViewEffect>> recordingBackgroundEffectObserver =
+      new RecordingObserver<>();
+
+  @Before
+  public void setUp() {
+    fakeLifecycle = new FakeLifecycleOwner();
+    recordedEvents.clear();
```

Comment:
Nit: maybe do this reinitialisation in a `tearDown` method instead?

Reply:
(applies up to line 73 I guess)

Reply:
Done.